## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445812`
Commit: `1500036ee09f52f110dc64e2975d224d61434687`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>mautrix-signal</li>
  </ul>
</details>
<details>
  <summary>:x: 24 packages failed to build:</summary>
  <ul>
    <li>libsignal-ffi</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python312Packages.ddgs</li>
    <li>python312Packages.ddgs.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.ddgs</li>
    <li>python313Packages.ddgs.dist</li>
    <li>python313Packages.duckduckgo-search</li>
    <li>python313Packages.duckduckgo-search.dist</li>
    <li>python313Packages.primp</li>
    <li>python313Packages.primp.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>signal-desktop</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>boringssl</li>
    <li>boringssl.bin</li>
    <li>boringssl.dev</li>
    <li>janus-gateway</li>
    <li>janus-gateway.dev</li>
    <li>janus-gateway.doc</li>
    <li>janus-gateway.man</li>
    <li>ustreamer</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>libsignal-ffi</summary>
<pre>3   | ...512 : i32 = 2054 ; pub const SSL_CURVE_SECP256R1 : i32 = 23 ; pub const SSL_CURVE_SECP384R1 : i32 = 24 ; pub const SSL_CURVE_SECP521R1...
    |                       ----------------------------------- similarly named constant `SSL_CURVE_SECP256R1` defined here

error[E0531]: cannot find unit struct, unit variant or constant `SSL_CURVE_SECP224R1` in crate `ffi`
   --> /build/libsignal-ffi-0.80.3-vendor/boring-4.18.0/src/ssl/mod.rs:777:18
    |
777 |             ffi::SSL_CURVE_SECP224R1 => Some(ffi::NID_secp224r1),
    |                  ^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_CURVE_SECP256R1`
    |
   ::: /build/source/target/x86_64-unknown-linux-gnu/release/build/boring-sys-2773a803d833ebfc/out/bindings.rs:3:253192
    |
3   | ...512 : i32 = 2054 ; pub const SSL_CURVE_SECP256R1 : i32 = 23 ; pub const SSL_CURVE_SECP384R1 : i32 = 24 ; pub const SSL_CURVE_SECP521R1...
    |                       ----------------------------------- similarly named constant `SSL_CURVE_SECP256R1` defined here

   Compiling zkgroup v0.9.0 (/build/source/rust/zkgroup)
   Compiling hyper v1.6.0
Some errors have detailed explanations: E0425, E0531.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `boring` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/bo/boringssl/package.nix
+++ b/pkgs/by-name/bo/boringssl/package.nix
@@ -8,16 +8,21 @@
   buildGoModule,
 }:
 
-# reference: https://boringssl.googlesource.com/boringssl/+/2661/BUILDING.md
-buildGoModule {
+# reference: https://boringssl.googlesource.com/boringssl/+/refs/tags/0.20250818.0/BUILDING.md
+buildGoModule (finalAttrs: {
   pname = "boringssl";
-  version = "unstable-2024-09-20";
+  version = "0.20250818.0";
 
   src = fetchgit {
     url = "https://boringssl.googlesource.com/boringssl";
-    rev = "718900aeb84c601523e71abbd18fd70c9e2ad884";
-    hash = "sha256-TdSObRECiGRQcgz6N2LhKvSi9yRYOZYJdK6MyfJX2Bo=";
+    tag = finalAttrs.version;
+    hash = "sha256-lykIlC0tvjtjjS/rQTeX4vK9PgI+A8EnasEC+HYspvg=";
   };
+
+  patches = [
+    # Add SECP224R1 for backward compatibility
+    ./secp224r1-compat.patch
+  ];
 
   nativeBuildInputs = [
     cmake
@@ -25,7 +30,7 @@ buildGoModule {
     perl
   ];
 
-  vendorHash = "sha256-GlhLsPD+yp2LdqsIsfXNEaNKKlc76p0kBCyu4rlEmMg=";
+  vendorHash = "sha256-IXmnoCYLoiQ/XL2wjksRFv5Kwsje0VNkcupgGxG6rSY=";
   proxyVendor = true;
 
   # hack to get both go and cmake configure phase
@@ -59,13 +64,10 @@ buildGoModule {
 
     mkdir -p $bin/bin $dev $out/lib
 
-    mv tool/bssl $bin/bin
+    install -Dm755 bssl -t $bin/bin
+    install -Dm644 {libboringssl_gtest,libcrypto,libdecrepit,libpki,libssl,libtest_support_lib}.a -t $out/lib
 
-    mv ssl/libssl.a           $out/lib
-    mv crypto/libcrypto.a     $out/lib
-    mv decrepit/libdecrepit.a $out/lib
-
-    mv ../include $dev
+    cp -r ../include $dev
 
     runHook postInstall
   '';
@@ -76,16 +78,16 @@ buildGoModule {
     "dev"
   ];
 
-  meta = with lib; {
+  meta = {
     description = "Free TLS/SSL implementation";
     mainProgram = "bssl";
     homepage = "https://boringssl.googlesource.com";
-    maintainers = [ maintainers.thoughtpolice ];
-    license = with licenses; [
+    maintainers = [ lib.maintainers.thoughtpolice ];
+    license = with lib.licenses; [
       openssl
       isc
       mit
       bsd3
     ];
   };
-}
+})

--- a/pkgs/by-name/bo/boringssl/secp224r1-compat.patch
+++ b/pkgs/by-name/bo/boringssl/secp224r1-compat.patch
@@ -1,0 +1,20 @@
+diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
+index 51417d412..a961a1093 100644
+--- a/include/openssl/ssl.h
++++ b/include/openssl/ssl.h
+@@ -2522,6 +2522,7 @@ OPENSSL_EXPORT size_t SSL_CTX_get_num_tickets(const SSL_CTX *ctx);
+ // |SSL_SIGN_*|.
+ 
+ // SSL_GROUP_* define TLS group IDs.
++#define SSL_GROUP_SECP224R1 22
+ #define SSL_GROUP_SECP256R1 23
+ #define SSL_GROUP_SECP384R1 24
+ #define SSL_GROUP_SECP521R1 25
+@@ -5836,6 +5837,7 @@ OPENSSL_EXPORT int SSL_CTX_set_tlsext_status_arg(SSL_CTX *ctx, void *arg);
+ #define SSL_R_TLSV1_CERTIFICATE_REQUIRED SSL_R_TLSV1_ALERT_CERTIFICATE_REQUIRED
+ 
+ // The following symbols are compatibility aliases for |SSL_GROUP_*|.
++#define SSL_CURVE_SECP224R1 SSL_GROUP_SECP224R1
+ #define SSL_CURVE_SECP256R1 SSL_GROUP_SECP256R1
+ #define SSL_CURVE_SECP384R1 SSL_GROUP_SECP384R1
+ #define SSL_CURVE_SECP521R1 SSL_GROUP_SECP521R1

--- a/pkgs/development/python-modules/primp/default.nix
+++ b/pkgs/development/python-modules/primp/default.nix
@@ -55,6 +55,21 @@ let
       );
 
     vendorHash = "sha256-06MkjXl0DKFzIH/H+uT9kXsQdPq7qdZh2dlLW/YhJuk=";
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $bin/bin $dev $out/lib
+
+      install -Dm755 tool/bssl -t $bin/bin
+      install -Dm644 ssl/libssl.a -t $out/lib
+      install -Dm644 crypto/libcrypto.a -t $out/lib
+      install -Dm644 decrepit/libdecrepit.a -t $out/lib
+
+      cp -r ../include $dev
+
+      runHook postInstall
+    '';
   });
   # boring-sys expects the static libraries in build/ instead of lib/
   boringssl-wrapper = runCommand "boringssl-wrapper" { } ''

